odoo/addons/base/models/ir_actions.py

```diff
@@ -251,6 +251,18 @@ def _get_readable_fields(self):
             "path",
         }
 
+    def _get_actions_by_type(self):
+        """
+        Returns:
+            A dict of {type: [action]} of actions in this recordset
+            mapped to their type. The actions lists are recordsets of the proper type.
+        """
+        types = self.mapped('type')
+        res = dict.fromkeys(types)
+        for action_type in types:
+            ids = self.filtered(lambda a: a.type == action_type).ids
```

Comment:
I would write that as a simple for loop and `res = defaultdict(set)   # map action_type => ids`

Reply:
I am convinced this function is useful in anycase, but I would write it in studio to avoid 2 PRs and altering base

Reply:
Why do we need to add this function? It seems unused.

```python
return {
  action_type: self.env[action_type].browse(actions.ids)
  for action_type, actions in self.grouped('type').items()
}
```